app/src/main/java/com/example/esha/movieapplication/Activities/FullMovieActivity.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    /**
+     * obtains trailer information of a particular movie and its reviews.
+     * displays the information in the users UI.
+     */
+
+    public void getData() {
+        Intent movieIntent = getIntent();
+        if (movieIntent == null) {
+            return;
+        }
+        final Result result = movieIntent.getParcelableExtra("result");
+        movieId = result.getId();
+
+        RetrofitManager.getInstance().getMovieTrailer(result.getId(), BuildConfig.TMDBMOVIEAPIKEY, new Callback<MovieVideoList>() {
+
+            @Override
+            public void onResponse(Call<MovieVideoList> call, Response<MovieVideoList> response) {
+
+                if (response.code() == 200) {
+                    trailerList.addAll(response.body().getResults());
+
+                } else {
+                    Toast.makeText(FullMovieActivity.this, " " + response.code(), Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<MovieVideoList> call, Throwable t) {
+                Toast.makeText(FullMovieActivity.this, t.getLocalizedMessage(), Toast.LENGTH_SHORT).show();
+                Log.i(TAG, "onFailure: " + t.getLocalizedMessage());
+
+            }
+        });
+
+        RetrofitManager.getInstance().getMovieReview(result.getId(), BuildConfig.TMDBMOVIEAPIKEY, new Callback<MovieReviewList>() {
+            @Override
+            public void onResponse(Call<MovieReviewList> call, Response<MovieReviewList> response) {
+                Log.i(TAG, "onResponse: " + result.getId());
+                if (response.code() == 200) {
+
+                    reviewList.addAll(response.body().getResults());
+                    reviewAdapter.notifyDataSetChanged();
+                } else {
+                    Toast.makeText(FullMovieActivity.this, "" + result.getId(), Toast.LENGTH_SHORT).show();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<MovieReviewList> call, Throwable t) {
+                Toast.makeText(FullMovieActivity.this, "" + result.getId(), Toast.LENGTH_SHORT).show();
+            }
+        });
+
+        Glide.with(FullMovieActivity.this)
+                .load("http://image.tmdb.org/t/p/w185//" + result.getPosterPath())
+                .into(imagePoster);
+        tvTitle.setText(result.getOriginalTitle() + " (" + result.getReleaaseDate().substring(0, 4) + ")");
+        ratingBar.setRating(result.getVoteAverage());
+        displayVoteAverage.setText("\t" + result.getVoteAverage() + "/10");
+        tvOverview.setText(result.getOverview());
+    }
+
+    /**
+     * initiates the youtube application to handle the request of the user to play the trailer of a particular movie.
+     * @param view
+     */
+    public void startIntent(View view) {
+        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.youtube.com/watch?v=" + trailerList.get(0).getKey()));
```

Comment:
@Esha-Rajbhandari what if position 0 is null or empty. Check for exceptions as well